keylime/cmd_exec.py

```diff
@@ -1,23 +1,37 @@
 import os
 import subprocess
 import time
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union, cast
 
 EXIT_SUCESS = 0
 
+EnvType = Dict[str, str]
 
-def _execute(cmd, env=None, **kwargs):
+
+def _execute(
+    cmd: Sequence[str], env: Optional[EnvType] = None, **kwargs
+) -> Tuple[Union[str, bytes], Union[str, bytes], int]:
```

Comment:
I think that all the communications are expected to be as strings. Maybe I am wrong, but if we pass `text=True` we can remove the Union

Reply:
Another one:
```
                        cmd = ("openssl", "crl", "-inform", "der", "-in", "cacrl.der", "-text", "-noout")
                        retout = cmd_exec.run(cmd)["retout"]
                        for line in retout:
                            line = line.strip()
                            if line.startswith(b"Next Update:"):
```

And all callers from tpm.py also currently get `bytes` back. So it be `bytes` then.

Reply:
Yes, that also would work. I suggested the str because all are doing decode, that I guess can be moved early. But I am fine with the reverse logic (produce bytes and decode as str later)